src/device/device_ledger.cpp

```diff
@@ -411,16 +429,55 @@ namespace hw {
       return true;
     }
 
+    bool  device_ledger::set_mode(unsigned int mode) {
+        AUTO_LOCK_CMD();
+
+        int offset;
+
+        reset_buffer();
+
+        switch(mode) {
+        case TRANSACTION_CREATE_REAL:
+        case TRANSACTION_CREATE_FAKE:
+          this->buffer_send[0] = 0x00;
+          this->buffer_send[1] = INS_SET_SIGNATURE_MODE;
+          this->buffer_send[2] = 0x01;
+          this->buffer_send[3] = 0x00;
+          this->buffer_send[4] = 0x00;
+          offset = 5;
+          //options
+          this->buffer_send[offset] = 0x00;
+          offset += 1;
+          //account
+          this->buffer_send[offset] = mode;
+          offset += 1;
+
+          this->buffer_send[4] = offset-5;
+          this->length_send = offset;
+          this->exchange();
+
+          this->mode = mode;
+          break;
+
+        case TRANSACTION_PARSE: 
+        case NONE:
+          this->mode = mode;
+          break;
+        default:
+           CHECK_AND_ASSERT_THROW_MES(false, " device_ledger::set_mode(unsigned int mode): invalid mode: "<<mode);
+        }
+        MDEBUG("Switch to mode: " <<mode);
+        return true;
+    }
+
 
     /* ======================================================================= */
     /*                             WALLET & ADDRESS                            */
     /* ======================================================================= */
 
-    /* Application API */
-     bool device_ledger::get_public_address(cryptonote::account_public_address &pubkey){
+     bool device_ledger::get_public_address(cryptonote::account_public_address &pubkey){  
```

Comment:
Please try to avoid spurious changes (especially ones that just add whitespace at eol :P)

Reply:
I try, I try... :)